collector/src/main/java/io/prometheus/jmx/JmxScraper.java

```diff
@@ -266,7 +267,51 @@ private void processBeanValue(
                 }
             }
         } else if (value.getClass().isArray()) {
-            logScrape(domain, "arrays are unsupported");
+            logScrape( domain, "arrays are unsupported" );
+        } else if (value instanceof Map ) {
+            try {
+                Map map = ( Map ) value;
+                for ( Object key : map.keySet() ) {
+                    attrKeys = new LinkedList<String>(attrKeys);
+                    attrKeys.add(key.toString());
+                    processBeanValue(
+                            domain,
+                            beanProperties,
+                            attrKeys,
+                            key.toString(), // forcing it to be a string
+                            "string",
+                            "hashmap",
+                            map.get( key ) ); // we don't go deeper into map itself
+                    attrKeys.removeLast();
+                }
+            } catch ( Exception e ) {
+                logScrape( domain + beanProperties, attrType + "had exception" );
+            }
+        } else if (value instanceof Object ) { // always is going to be true for all objects anyway
+            logger.info( value.getClass().getDeclaredMethods().toString() );
+            for (Method method : value.getClass().getDeclaredMethods() ) {
+                try {
+
+                    String name = method.getName();
+                    if ( name.startsWith( "get" )  || name.startsWith( "is" )) {
```

Comment:
I don't think we should automatically call functions, a lot of them are going to change state and/or be slow.